google/cloud/storage/testing/temp_file.cc

```diff
@@ -27,16 +28,14 @@ namespace testing {
 TempFile::TempFile(std::string const& content) {
   // This is obviously racy, but there is no portable way to create a
   // uniquely-named temporary file and know its name.
-  char tmpfile_name[L_tmpnam];
-  if (!std::tmpnam(tmpfile_name)) {
-    Terminate("Failed to create a temporary file name");
-  }
-  std::ofstream f(tmpfile_name, std::ios::binary | std::ios::trunc);
+  auto generator = google::cloud::internal::MakeDefaultPRNG();
+  auto name = MakeRandomFileName(generator);
+  std::ofstream f(name, std::ios::binary | std::ios::trunc);
```

Comment:
The process's CWD may not be writable, so we should probably place `name` in an appropriate temp directory for the platform. We may be able to do this with a gtest function `testing::TempDir() + "/" + name` or something.